includes/wikia/tasks/Tasks/UserRegistrationTask.class.php

```diff
@@ -0,0 +1,25 @@
+<?php
+namespace Wikia\Tasks\Tasks;
+
+use Hooks;
+use User;
+use WebRequest;
+
+/**
+ * Class UserRegistrationTask notifies extensions (via AddNewAccount hook) on user registration
+ */
+class UserRegistrationTask extends BaseTask {
+	public function callUserRegistrationHooks( array $userRegistrationInfo ) {
+		list( $userName, $clientIp ) = $userRegistrationInfo;
+
+		/** @var WebRequest $wgRequest */
+		global $wgRequest;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)